src/xbot/common/logic/Latch.java

```diff
@@ -23,28 +23,28 @@ public void setValue(boolean value) {
         boolean different = this.value != value;
 
         if (different) {
+            this.value = value;
             switch (latchType) {
                 case Both:
-                    saveValueAndAlertWatchers(value);
+                    alertWatchers(value);
                     break;
                 case RisingEdge:
                     if (value) {
-                        saveValueAndAlertWatchers(true);
+                        alertWatchers(true);
                     }
                     break;
                 case FallingEdge:
                     if (!value) {
-                        saveValueAndAlertWatchers(false);
+                        alertWatchers(false);
                     }
                     break;
                 default:
-                    this.value = value;
                     break;
             }
         }
     }
 
-    private void saveValueAndAlertWatchers(boolean value) {
+    private void alertWatchers(boolean value) {
         EdgeType edgeType = value ? EdgeType.RisingEdge : EdgeType.FallingEdge;
```

Comment:
Can this method just accept an `EdgeType` so we don't need to extrapolate like this?

Reply:
Agreed - you do all the work to figure out the edge type in setValue, but then throw it away and figure it out again in alertWatchers. =]